File: csharp/src/Training.Patterns/Structural/Composite/Employee.cs
namespace Training.Patterns.Structural.Composite;

public class Employee : Node
{
    public Employee(string name) : base(name)
    {
    }

    public override void AddChild(Node node) => throw new NotSupportedException();

    public override void PrintInfo()
    {
        Console.WriteLine(" - Employee " + name);
    }
}

File: typescript/src/patterns/structural/composite/Employee.ts
import { UnsupportedOperationError } from '../../../shared/errors.js';
import { Node } from './Node.js';

export class Employee extends Node {
  constructor(name: string) {
    super(name);
  }

  override addChild(_node: Node): void {
    throw new UnsupportedOperationError();
  }

  printInfo(): void {
    console.log(` - Employee ${this.name}`);
  }
}
